vmm/src/lib.rs

```diff
@@ -410,34 +410,34 @@ impl Vmm {
                                         continue;
                                     }
 
-                                    let response = match self.vm_boot() {
-                                        Ok(_) => Ok(ApiResponsePayload::Empty),
-                                        Err(e) => Err(ApiError::VmBoot(e)),
-                                    };
+                                    let response = self
+                                        .vm_boot()
+                                        .map_err(ApiError::VmBoot)
+                                        .map(|_| ApiResponsePayload::Empty);
 
                                     sender.send(response).map_err(Error::ApiResponseSend)?;
                                 }
                                 ApiRequest::VmShutdown(sender) => {
-                                    let response = match self.vm_shutdown() {
-                                        Ok(_) => Ok(ApiResponsePayload::Empty),
-                                        Err(e) => Err(ApiError::VmShutdown(e)),
-                                    };
+                                    let response = self
+                                        .vm_shutdown()
+                                        .map_err(ApiError::VmShutdown)
+                                        .map(|_| ApiResponsePayload::Empty);
```

Comment:
We're loosing error propagation here.
Applies for multiple cases on this commit.

Reply:
I may be missing your point, so let's take the `vm_shutdown()` example:

If `vm_shutdown()` fails, it will return a `err: Error(VmError)`, that we'll map to an `ApiError::VmShutdown(err)`. So in that case `response` is a `Error(ApiError::VmShutdown(VmError))` and afaict the error is propagated.

I verified this by forcing a `VmError::NotCreated` out of vm_shutdown, simulating a VM shutdown error. When I call into the `vm.shutdown` endpoint I get the following HTTP response back:

```
$ ./vm-shutdown.sh /tmp/cloud-hypervisor.sock 
HTTP/1.1 500 
Server: Firecracker API
Connection: keep-alive
Content-Type: text/plain
Content-Length: 36

VmShutdown(VmShutdown(VmNotCreated))
```

Reply:
Yes sorry I missed that. All good then :)